src/main/java/com/wojciechbarwinski/demo/legendary_warehouse/stock/StockRepositoryImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.wojciechbarwinski.demo.legendary_warehouse.stock;
+
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.stereotype.Repository;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Getter
+@Setter
+@Repository
+public class StockRepositoryImpl implements StockRepository {
+
+    private Map<String, StockItem> productsMap;
+
+    private StockRepositoryImpl() {
+        productsMap = new ConcurrentHashMap<>();
```

Comment:
Czemu Concurrent? :D

Reply:
specjalnie to tak jakos zaznaczyłem bo... a co jak beda 2 zamowienia na raz sie procedowały i miedzy sprawdzeniem a usunięciem z magazynu, inne zamowienie dokona sprawdzenia? 
Chodziłomi aby w ogóle ten watek ... wielowątkowy poruszyć i czy mógłbyś mi cokolwiek z nim wyjaśnić.